src/psyclone/psyir/nodes/array_reference.py

```diff
@@ -187,6 +187,12 @@ def datatype(self):
             # shape in the fparser2 parse tree but, at this point, we
             # wouldn't know what the variable name should be (TODO #2137).
             return UnresolvedType()
+        if not isinstance(self.symbol.datatype, ArrayType):
+            # If we have an array reference to a symbol that is not considered
+            # an array by PSyIR, we don't know how to retrieve its element type
+            # (TODO #2448)
+            return UnresolvedType()
```

Comment:
This line is missing from CodeCov (what fails is the Github connector but you can see the results at https://app.codecov.io/gh/stfc/PSyclone/pull/2511).

The ArrayReference create method will not let you build such a reference, but you can take one of the existing references   in `src/psyclone/tests/psyir/nodes/array_reference_test.py::test_array_datatype`, replace its .symbol to a new DataSymbol of StructureType, and call the .datatype again to test it.